pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -171,6 +171,21 @@ def __init__(self, filepath: Optional[str] = None, monitor: str = 'val_loss', ve
 
         self.kth_value, self.mode = mode_dict[mode]
 
+    def _validate_condition_metric(self, logs):
+        monitor_val = logs.get(self.monitor)
+
+        if monitor_val is None:  # self.monitor wasn't found in the metrics dict
+            if self.verbose > 0:
+                error_msg = (f'Checkpointing conditioned on metric `{self.monitor}`'
+                             f' which is not available. Either add `{self.monitor}` to the return of'
+                             f' validation_epoch end or modify your CheckPoint callback to use any of the'
+                             f' following: `{"`, `".join(list(logs.keys()))}`')
+                rank_zero_warn(error_msg, RuntimeWarning)
+
+            return False
+
+        return True
```

Comment:
why do you define return if it is not used in the followed-up call?

Reply:
yeah not required